examples/hikari_tanjun_basic_queue/music.py

```diff
@@ -0,0 +1,351 @@
+import typing
+
+import hikari
+import lavasnek_rs
+import tanjun
+
+
+music = tanjun.Component()
+
+
+@music.with_slash_command
+@tanjun.as_slash_command("join", "Connect the bot to a voice channel.")
+async def join_as_slash(
+    ctx: tanjun.abc.SlashContext,
+    lavalink: lavasnek_rs.Lavalink = tanjun.injected(type=lavasnek_rs.Lavalink),
+) -> None:
+    await _join_voice(ctx, lavalink)
+
+
+@music.with_message_command
+@tanjun.as_message_command("join")
+async def join_as_message(
+    ctx: tanjun.abc.MessageContext,
+    lavalink: lavasnek_rs.Lavalink = tanjun.injected(type=lavasnek_rs.Lavalink),
+) -> None:
+    """Connect the bot to a voice channel."""
+    await _join_voice(ctx, lavalink)
+
+
+async def _join_voice(ctx: tanjun.abc.Context, lavalink: lavasnek_rs.Lavalink) -> int:
+    """Joins your voice channel."""
+    assert ctx.guild_id is not None
+
+    if ctx.client.cache and ctx.client.shards:
+        # Get the users voice state if they didnt pass a channel
+        if not (voice_state := ctx.client.cache.get_voice_state(ctx.guild_id, ctx.author)):
+            await ctx.respond("Please connect to a voice channel.")
+            return 1
+
+        # Make sure our user is cached
+        elif own_user := ctx.client.shards.get_me():
+            # Check if there is a voice channel for the author
+            if voice_state.channel_id and own_user.id in (
+                # Check if we are in that channel
+                ctx.client.cache.get_voice_states_view_for_channel(
+                    ctx.guild_id,
+                    voice_state.channel_id,
+                )
+            ):
+                # We are in the channel already
+                await ctx.respond(f"I am already connected to <#{voice_state.channel_id}>.")
+                return 1
```

Comment:
To be consistent with the other examples, this should join the new channel regardless. It's a basic example, so this is meant to show capabilities of the library, not how someone should make their bot.
In this case, it would show the ability for lavasnek to keep playing music when switching channels.

Reply:
This will still join a new voice channel, it just won't join the same voice channel that its already in and say it joined like the other examples do.